e2e/test/scenarios/actions/actions-in-object-detail-view.cy.spec.js

```diff
@@ -185,6 +187,38 @@ describe(
         });
       },
     );
+
+    it("should show detailed form errors for constraint violations when executing model actions", () => {
+      const actionName = "Update";
+
+      cy.signInAsAdmin();
+
+      cy.get("@modelId").then(modelId => {
+        cy.visit(`/model/${modelId}/detail`);
+        cy.wait("@getModel");
+        createImplicitActions({ modelId });
+        visitObjectDetail(modelId, FIRST_SCORE_ROW_ID);
+        openUpdateObjectModal();
+      });
+
+      actionExecuteModal().within(() => {
+        cy.wait("@prefetchValues");
+
+        actionForm().within(() => {
+          cy.findByLabelText("Team Name").clear().type("Dusty Ducks");
+          cy.findByText(actionName).click();
+        });
+
+        cy.wait("@executeAction");
+
+        cy.findByLabelText("Team Name").should("not.exist");
+        cy.findByLabelText(
+          "Team Name: This Team_name value already exists.",
+        ).should("exist");
```

Comment:
What do you think about swapping the assertions so we don't need to wait for `executeAction`?

```suggestion
        cy.findByLabelText(
          "Team Name: This Team_name value already exists.",
        ).should("exist");
        cy.findByLabelText("Team Name").should("not.exist");
```

By doing this we ensure `"Team Name: This Team_name value already exists."` exist that means it waits for the API to finish before asserting that `Team Name` label has gone (since it's replaced by the one with the error message) This one would eliminate the need to intercept `executeAction`

Reply:
> By doing this we ensure `"Team Name: This Team_name value already exists."` exist that means it waits for the API to finish before asserting that `Team Name` label has gone (since it's replaced by the one with the error message)

I think it's pretty much equivalent because `Team Name` will be gone only if the other one replaces it.

> This one would eliminate the need to intercept `executeAction`

Currently the order of these lines reflects the order of how things are happening on the webpage - I believe this practice helps with flakiness.

`cy.wait("@executeAction")` has some benefits:
- it's explicit
- it has different timeout (5s) than `.should("exist")` (4s)
- so in total we get 9s of timeout instead of just 4s (helps with flakiness)

To sum up: I'd keep it as is.

Reply:
Also see: https://github.com/metabase/metabase/pull/32682#discussion_r1275523774